public/app/features/alerting/unified/utils/matchers.ts

```diff
@@ -144,6 +145,22 @@ export function quoteWithEscape(input: string) {
   return `"${escaped}"`;
 }
 
+/**
+ * Quotes string only when reserved characters are used
+ */
+const RESERVED_CHARACTERS = /[\{\}\!\=\~\,\\\"\'\`\s]+/;
```

Comment:
What do you think about this? (It was fiddly to get exactly the same but it makes the characters potentially a little easier to read?)

```ts
const RESERVED_CHARACTERS = ['{', '}', '!', '=', '~', ',', '\\', '"', "'", '`', /\s/];
const RESERVED_CHARACTERS_REGEX = new RegExp(`[\\${RESERVED_CHARACTERS.join('\\')}]+`);
```

Reply:
I'll let Gilles answer too, but for me I'm not sure it's worthwhile? Here are my thoughts:

1. The reserved characters are well documented in Alertmanager and will probably never change (without having to do another migration) as it's a breaking change to the grammar. I would be very surprised if you make changes to this regex in future.
2. I did ~2 hours of manual testing last night in dev using the existing regex and I'm confident in it 😄 If possible, I'd like to not have to do it again unless there is a really good reason for it.

Reply:
Sure, valid points :+1: I agree that it's unlikely we'd change the regex, but if it's nicer to read then its clearer what we're trying to do here.

This generated regex is the same as the one from before - it hasn't changed and the (unit) tests all pass with the different approach (with a small change to just make the formatting happier with the `\s`):

```ts
const RESERVED_CHARACTERS = ['{', '}', '!', '=', '~', ',', '\\', '"', "'", '`'];
const RESERVED_CHARACTERS_REGEX = new RegExp(`[\\${RESERVED_CHARACTERS.join('\\')}\\s]+`);
const original = /[\{\}\!\=\~\,\\\"\'\`\s]+/;

console.log(RESERVED_CHARACTERS_REGEX.toString() === original.toString()) // true
```

However - I won't die on this hill at all 😁

Reply:
OK! This makes me feel better 😄 Gilles, up to you 👈

Reply:
I'll keep the regular expression, but I'll add a code comment that will help with deconstructing the array of special characters we have to pay attention to 👍